### PR TITLE
DataGen Take 2: Using the Ultimate Key Ring 

### PR DESCRIPTION
**Pull request by Dragon-Seeker**
PR to start the ground work for other Data Providers and the providers needed for Block and Item Tag Generation

### CODE REVIEW

**Comment by TropheusJ** [comment]
yeah I think I'm gonna need some explanations
image
image

**Comment by Stereo528** [comment]
A lot of those are JSON files by the looks of it...

**Comment by Dragon-Seeker** [comment]
> 

What do you expect when I was told to go crazy with making tags?

File: fabric/src/generated/resources/data/c/tags/blocks/material/crimson_wart.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/crimson_wart"
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/blocks/material/crimson_wart.json:4]: "minecraft:nether_wart_block" should also be included here

File: fabric/src/generated/resources/data/c/tags/blocks/material/dripstone.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/dripstone"
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/blocks/material/dripstone.json:4]: "minecraft:dripstone_block"

File: fabric/src/generated/resources/data/c/tags/blocks/material/netherrack.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/netherrack",
+    "minecraft:netherrack"
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/blocks/material/netherrack.json:5]: "minecraft:nether_bricks" and "minecraft:chiseled_nether_bricks" are also part of this set

File: fabric/src/generated/resources/data/c/tags/blocks/material/purpur.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/purpur",
+    "minecraft:purpur_pillar"
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/blocks/material/purpur.json:5]: "minecraft:purpur_block"

File: fabric/src/generated/resources/data/c/tags/blocks/material/terracotta/terracotta_bricks_block.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/terracotta/terracotta_bricks_block"
+  ]
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/blocks/material/terracotta/terracotta_bricks_block.json:5]: "minecraft:bricks"

File: fabric/src/generated/resources/data/c/tags/blocks/material/tinted_glass.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/tinted_glass"
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/blocks/material/tinted_glass.json:4]: "minecraft:tinted_glass"

File: fabric/src/generated/resources/data/c/tags/items/material/blackstone.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  "replace": false,
+  "values": [
+    "#consistency_plus:material/blackstone",
+    "minecraft:polished_blackstone",
+    "minecraft:blackstone_slab",
+    "minecraft:blackstone_stairs",
+    "minecraft:blackstone",
+    "minecraft:blackstone_wall",
+    "minecraft:polished_blackstone_wall",
+    "minecraft:polished_blackstone_slab",
```
Comment (Cart3r1234) [inline fabric/src/generated/resources/data/c/tags/items/material/blackstone.json:11]: missing vanilla's blackstone bricks due to "polished_blackstone_bricks" being the ID

**Comment by Cart3r1234** [comment]
Aside from the few things I commented on, I think it's ready.

**Comment by Cart3r1234** [comment]
Also, make sure it addresses #96 and adds our items (copper/netherite nuggets, netherite horse armor, warped wart plant, withered bones/bone meal, turtle armor, and brick items) to the respective tags if they exist.

**Comment by Dragon-Seeker** [comment]
> Also, make sure it addresses #96 and adds our items (copper/netherite nuggets, netherite horse armor, warped wart plant, withered bones/bone meal, turtle armor, and brick items) to the respective tags if they exist.

Done

**Comment by Siuolthepic** [comment]
LGETM, I approve of this

**Pull request merged**